worlds/rogue_legacy/Rules.py

```diff
@@ -1,37 +1,55 @@
-from BaseClasses import MultiWorld, CollectionState
+from BaseClasses import MultiWorld
 
 from ..AutoWorld import LogicMixin
 from ..generic.Rules import set_rule
 
 
-class LegacyLogic(LogicMixin):
-    def has_any_vendors(self: CollectionState, player: int) -> bool:
-        return self.has_any({"Blacksmith", "Enchantress"}, player)
-
-    def has_all_vendors(self: CollectionState, player: int) -> bool:
-        return self.has_all({"Blacksmith", "Enchantress"}, player)
-
-    def has_stat_upgrades(self, player: int, amount: int) -> bool:
-        return self.stat_upgrade_count(player) >= amount
-
-    def total_stat_upgrades_count(self, player: int) -> int:
+class RLLogic(LogicMixin):
+    def _get_upgrade_total(self, player: int) -> int:
```

Comment:
all logic mixins get mashed together into one class, so if any other games has a _get_upgrade_total it'll clash.
Might be best to add a prefix like `rl_` to these when changing them anyway.

Reply:
TIL. Removed Mixin.